fv/wireguard_test.go

```diff
@@ -348,6 +335,52 @@ var _ = infrastructure.DatastoreDescribe("WireGuard-Supported", []apiconfig.Data
 			Expect(rcvd[1]).NotTo(BeEmpty())
 			Expect(rcvd[1]).To(Equal(sent[0]))
 		})
+
+		It("between pod to pod should be encrypted using wg tunnel with egress policy applied", func() {
+			policy := api.NewGlobalNetworkPolicy()
+
+			policy.Name = "f01-egress-deny"
+			policy.Spec.Egress = []api.Rule{{Action: api.Deny}}
+			policy.Spec.Selector = fmt.Sprintf("name in { '%s', '%s'}", wls[0].Name, wls[1].Name)
+			_, err := client.GlobalNetworkPolicies().Create(utils.Ctx, policy, utils.NoOptions)
+			Expect(err).NotTo(HaveOccurred())
+
+			cc.ExpectNone(wls[0], wls[1])
+			cc.ExpectNone(wls[1], wls[0])
+			cc.CheckConnectivity()
+
+			By("verifying tunnelled packet count is non-zero")
+			for i := range felixes {
+				Consistently(func() int {
+					return tcpdumps[i].MatchCount("numInTunnelPackets")
+				}, "5s", "100ms").Should(BeNumerically("==", 0))
+				Consistently(func() int {
+					return tcpdumps[i].MatchCount("numOutTunnelPackets")
+				}, "5s", "100ms").Should(BeNumerically("==", 0))
+			}
+
+			cc.ResetExpectations()
+
+			policy.Name = "f01-egress-allow"
```

Comment:
I think this relies on the policy ordering being alphanumeric - which we don't actually expose anywhere.

I would either:
-  Modify the original policy, or
-  Set the Order number so this is definitely lower order than the first.

Reply:
added order